inductiva/resources/machines.py

```diff
@@ -0,0 +1,107 @@
+"""Containes Machines class to manage resources."""
+
+from absl import logging
+from typing import Optional
+from uuid import UUID
+import uuid
+import inductiva
+from inductiva import api
+
+from inductiva.client import ApiClient, ApiException
+from inductiva.client.apis.tags.instance_api import InstanceApi
+from inductiva.client.model.instance import Instance
+from inductiva.client.model.instance_group import InstanceGroup
+
+
+class Machines():
+    """Class to Google Cloud manage resources."""
+
+    def __init__(
+        self,
+        machine_type: str,
+        executer: str,
+        num_machines: int = 1,
+        spot: bool = False,
+        disk_size_gb: int = 30,
+        label: str = None,
+        resource_pool_id: Optional[UUID] = None,
+    ) -> None:
+        """Initialize the Machines class.
+
+        Args:
+            machine_type: The type of GC machine to launch. Ex: "e2-medium".
+            executer: The type of executer to launch. Ex: "gromacs", "openfoam".
+            num_machines: The number of machines to launch.
+            spot: Whether to use spot instances.
+            disk_size_gb: The size of the disk in GB. (min. 30 GB)
+            label: The label to assign to the machine group.
+            resource_pool_id: The resource pool ID to use. A new resource pool
+              id can be created using the
```

Comment:
It's unclear why we tell the user that a resource pool id can be created by invoking that method (does that mean that the user should worry about doing it?)... Probably what you meant was to specify the default behaviour if None is passed, right? That's a good idea. Rephrase to say "If None is provided, a unique id will be automatically generated.... I don't think we need to say how... but if so, say something like "... by invoking inductiva.resources.new_machine_group_id() " ....

Reply:
Since I we decided that users will create the machines by calling `inductiva.resources.launch_machine_group` (which is in the other script), that's where this id is automatically created and is passed here. So users will have to create a new id, only if they wanted to create a machine group directly from this object. Which raises a question, do we even need `inductiva.resources.launch_machine_group` method or should we just let them use this object to create resources?

Reply:
I think we settled on not having that method. And instead use the machine_group object with methods start() and terminate()...